### ISSUE TITLE
2.4 app creation crash on image load

### ISSUE THREAD

**Issue by ylluminate** [dmarmor/epichrome#308]
While creating a new app I received a crash while loading an image.  It appears that it accepted the image and app creation went on through completion, but this crash occurred in the middle of the process:
https://gist.github.com/ylluminate/fca147422e5599b73451bfc2c44c7037

I suspected it has something to do with an image selected for the icon, but since it was properly set I'm not sure...

Here's the image, just in case:
https://p153.p0.n0.cdn.getcloudapp.com/items/7KuPJYEJ/2b199b1c-e606-4bd1-a827-67b3d7dc5687.png

**Comment by dmarmor** [comment]
Can you post the log for this run of Epichrome? It'll be in `~/Library/Application Support/Epichrome/Logs/Epichrome`.

Thanks for sending the image too. I'll give it a try on my system and see if I can reproduce this.

**Comment by ylluminate** [comment]
I'm sorry, I missed this reply in my email (had a family death recently that has created a lot of challenges).  The log I believe for that date is this: 
* https://p153.p0.n0.cdn.getcloudapp.com/items/rRuGlvNO/c39ff848-99df-4fcb-8c3c-f3687d45baf1.txt

I just had this crash occur again on a new app creation. It happened at exactly the same point upon selecting an image. 
* Screenshot of the next step following crash: https://p153.p0.n0.cdn.getcloudapp.com/items/nOuoepb5/d7bcfb7d-f3b7-4c87-ac8c-7eba66f9dfe1.jpg
* Log at time of crash (before hitting "Confirm" in the screenshot above): https://p153.p0.n0.cdn.getcloudapp.com/items/E0uYz8Xm/a0f0904a-7166-4f4e-880e-7afebc17251b.txt
* Actual image (name: `mightycall.png`): https://p153.p0.n0.cdn.getcloudapp.com/items/04uAY0dq/3ae40eaf-ac07-4487-9e75-97dea9a4790f.png
* Crash report: https://p153.p0.n0.cdn.getcloudapp.com/items/rRuGlvDA/daa76916-3b43-43e7-aa80-0127919d52f0.txt

**Comment by dmarmor** [comment]
This is helpful. It looks like its the progress bar app that's crashing. Am I correct that the icon is properly created and the app creation works OK? It's just that the progress bar app crashes? Or does it screw up the whole app-creation process?

It's hard for me to figure out why the progress bar is crashing, as it works fine on my system for both of the icon sources you've sent me. I wonder if there's some problem on your system with Platypus (which is what app launchers and the progress bar app use). This could also possibly explain some of the problems you've been having with apps getting left activated after quit.

I'm sorry to hear about the death in your family, and hope you and your loved ones are recovering as best you can...

**Comment by ylluminate** [comment]
Yes, the icon is properly created and set and the app works seemingly fine. I have to assume that your assumption and understanding is correct here since all seems to work aside from that strange intermediary crash.

Hmm, perhaps you are right in that something else is causing the crash.  Can we add some verbosity in logging for this process that perhaps we can use to track this down further to something more specific?  

Thank you, yes, the death has been extremely difficult to deal with for all of us and it's probably only something that can ease with the passage of much time.

**Comment by dmarmor** [comment]
There's not a lot more verbosity I can provide, since the problem seems to be a fatal crash in a process I didn't write (Platypus), and it's not one that I can reproduce on my system. I'll see if there are some extra log messages I can throw into the icon-preview process in a future version to see if that will help.

**Comment by ylluminate** [comment]
As I noted, another option might be to manually execute each step via the shell (sans GUI) in order to see what output occurs at each step. Would that be possible?

**Comment by dmarmor** [comment]
That won't help in this case. The issue is happening in the creation of the icon preview, which is entirely inside one step of the process. The best bet will be to see if I can put a few more debugging messages into that code and see where it is when it dies.

**Comment by dmarmor** [comment]
So I've added a bunch of extra debugging messages during icon preview and creation in version 2.4.5, which should be up later today. If you can run Epichrome with debugging on and reproduce this crash and send me the log, I can try to start figuring out what's going on.
